CMakeLists.txt

```diff
@@ -53,9 +53,13 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 # Synthesize clang-resource-headers target if necessary.
 if (NOT TARGET clang-resource-headers)
+  find_program(CLANG_EXECUTABLE NAMES clang${LLVM_VERSION_MAJOR} clang-${LLVM_VERSION_MAJOR} clang REQUIRED)
```

Comment:
I think it's a little risky to include `clang` here -- it might refer to a version of Clang that is not compatible with the version we're linking with.

Maybe for this to work it would be safer to try and check that `clang --version` matches `LLVM_VERSION` somehow?

Reply:
It's a little uncomfortable that we're jumping outside the CMake-published config entirely trying to find headers that might match. I wonder if:

* LLVM/Clang's CMake modules contain a variable for the resource dir already
* ... and if not, whether it would be possible to get a patch in there to add it?

Reply:
> LLVM/Clang's CMake modules contain a variable for the resource dir already

LLVM-internal projects already have access to [get_clang_resource_dir](https://github.com/llvm/llvm-project/blob/main/cmake/Modules/GetClangResourceDir.cmake#L9). But other projects are expected to get this information from clang.

More info: https://github.com/llvm/llvm-project/tree/main/cmake#what-goes-here

Reply:
Yes, that page describes the current state. But I wonder if a case could be made to move `get_clang_resource_dir` into the "installed modules" in Clang/LLVM's CMake system.

I still feel like using `clang*` to probe is likely to fail, e.g. with:

* `clang` command is not installed, only libraries
* we accidentally pick the wrong `clang` and get a resource dir that's not compatible with libraries
* we're missing some weird `clang` decoration and don't find any binary

So it would be nice to push the challenge upstream :)

Reply:
I opened https://github.com/llvm/llvm-project/issues/95747 to see if it's possible to do anything on the Clang side (I'm not too good with CMake, so I don't have a clear idea how hard this is).

Reply:
Also raised this on the LLVM forums to see if someone bites: https://discourse.llvm.org/t/export-clang-resource-dir-from-clangconfig-cmake/80000